main.py

```diff
@@ -15,27 +14,101 @@ class Robot:
+        instead.
+        """
         opts = RobotClient.Options(
             refresh_interval=0,
             dial_options=DialOptions(credentials=secrets.creds)
         )
         self.robot = await RobotClient.at_address(secrets.address, opts)
-        self.servo = Servo.from_robot(self.robot, "servo")
+        self._servo = Servo.from_robot(self.robot, "servo")
+
+        self._mutex = asyncio.Lock()
+        self._count = 0  # Number of hands currently raised
+
+        # self._wiggler will become an asyncio.Task when the hand is raised. It
+        # will wiggle the hand when it has been raised for over
+        # INACTIVITY_PERIOD_S seconds.
+        self._wiggler = None
+
+    def _start_wiggler(self):
+        self._wiggler = asyncio.create_task(self._wiggle_on_inactivity())
+
+    async def _stop_wiggler(self):
+        self._wiggler.cancel()
+        await self._wiggler
+        self._wiggler = None
+
+    async def _wiggle_on_inactivity(self):
+        """
+        This is a background coroutine that wiggles the hand every
+        INACTIVITY_PERIOD_S seconds. It is started when the count of raised
+        hands becomes nonzero, and canceled when the count goes back to 0.
+        """
+        try:
+            while True:
+                await asyncio.sleep(self.INACTIVITY_PERIOD_S)
+                await self._wiggle_hand()
+        except asyncio.CancelledError:
+            return
 
     async def raise_hand(self):
-        await self.servo.move(self.UPPER_POSITION)
+        """
+        Call this to consider 1 extra person in the audience to have raised
+        their hand. If this is the first person to do so, we'll raise our
+        servo, and otherwise we take no action.
+        """
+        async with self._mutex:
+            self._count += 1
+            if self._count == 1:
+                await self._servo.move(self.UPPER_POSITION)
+                self._start_wiggler()
 
     async def lower_hand(self):
-        await self.servo.move(self.LOWER_POSITION)
+        """
+        Call this to consider 1 extra person in the audience to have lowered
+        their hand. If this is the last person who had their hand raised, we'll
+        lower our servo, and otherwise take no action.
+        """
+        async with self._mutex:
+            self._count -= 1
+            if self._count == 0:
+                await self._stop_wiggler()
+                await self._servo.move(self.LOWER_POSITION)
+
+    async def set_count(self, new_value):
+        """
+        Call this to set the number of hands raised in the audience to a certain
+        value. This is mainly used to "reset" the count of raised hands if
+        someone forgets to lower their hand.
+        """
+        async with self._mutex:
+            should_start_wiggler = self._count == 0 and new_value > 0
+            self._count = new_value
+            if self._count == 0:
+                await self._stop_wiggler()
```

Comment:
If the old count was 0 and the new count is also 0, we're gonna have a bad time because we'll `await self._wiggler` but that's really an `await None`. Serves me right for writing code I couldn't try out! I'll fix this next week.